encoder/vaapiencoder_h264.cpp

```diff
@@ -1142,10 +1326,17 @@ bool  VaapiEncoderH264::pictureReferenceListSet (
 
     for (i = 0; i < m_refList.size(); i++) {
         assert(picture->m_poc != m_refList[i]->m_poc);
-        if (picture->m_poc > m_refList[i]->m_poc)
+        if (picture->m_poc > m_refList[i]->m_poc
```

Comment:
It is a common logic for bot svt-t and non-svct case. One principle of this patch is make it common for svc-t and non-svct as possible as I can, because svc-t is still a AVC mode.